### PR TITLE
Provide Try{Into|From}Ctx impls for fixed byte array

### PR DESCRIPTION
**Pull request by nyurik**
This is a squashed #86.  I do not think this should be merged as is because it has a few reverts and a few other unrelated things.

### CODE REVIEW

**Comment by nyurik** [comment]
I just pushed a new change to this PR - which tries to fix a few merge conflicts.

@m4b - I do not think the macro rules should be changed. @Frostie314159 - did you intend to change those?

File: tests/api.rs [hunk 1]
```diff
@@ -1,3 +1,4 @@
+#![allow(clippy::disallowed_names)]
```
Comment (nyurik) [inline tests/api.rs:1]: i'm not sure this is needed.

**Comment by m4b** [comment]
I'm sorry I'm getting very confused with what's going on with all these PRs; I think at this stage we have to have the simple PR with a single commit that just adds the impls for fixed byte arrays; I don't want to review more diffs with unrelated changes, clippy changes, etc.

**Comment by nyurik** [comment]
@m4b agree - I also proposed this in #86. I created this PR only to help @Frostie314159 to see what his PR actually changes compared with the main branch.  There is no need to merge it.

**Comment by m4b** [comment]
i see, thank you @nyurik , i'm going to go to sleep now, it's late for me, thanks for both of your hard work and PRs, much appreciated :)

**Comment by Frostie314159** [comment]
Thanks for the help!

**Comment by m4b** [comment]
this is done in another pr

**Pull request closed without merging**